File: greedy/구명보트.py
def solution(people, limit):
    people.sort()
    couples = 0

    l = 0
    r = len(people) - 1

    while l < r:
        if people[l] + people[r] <= limit:
            l += 1
            couples += 1
        r -= 1

    return len(people) - couples

File: src/greedy/구명보트.java
package greedy;

import java.util.*;

public class 구명보트 {

    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int i = 0, j = people.length - 1;
        for (; i < j; j--) {
            if (people[i] + people[j] <= limit)
                i++;
        }
        return people.length - i;
    }
}
